### PR TITLE
[WIP] `stale-balance` command: force-sync EigenPods with slashed validators

### PR DESCRIPTION
**Pull request by jbrower95**
# stale-balance

- If an EigenPod has been slashed, you can use `stale-balance` to start a checkpoint against the EigenPod. We suggest avoiding doing this if the deviated amount is small (<5%), but it is a legal operation within the protocol at any time as a way to sync the beacon state with the execution state, when we know that a beacon slashing has occurred.

- To use;

`./cli stale-balance --podAddress $EIGENPOD_ADDRESS --execNode $NODE_ETH --beaconNode $NODE_BEACON --sender $EIGENPOD_PK --slashedValidatorIndex $SLASHED_VALIDATOR_INDEX`

This will prompt to complete any in-flight checkpoints, and then call verifyStaleBalances.

### CODE REVIEW

File: cli/main.go [hunk 1]
```diff
@@ -128,6 +157,76 @@ func main() {
 		EnableBashCompletion:   true,
 		UseShortOptionHandling: true,
 		Commands: []*cli.Command{
+			{
+				Name:  "stale-balance",
+				Args:  true,
+				Usage: "If needed, calls `verifyStaleBalances` to correct the balance on a pod you don't own. This will attempt to (1) conclude any existing checkpoint on the pod, and then (2) invoke verifyStaleBalances() to start another, more up-to-date, checkpoint.",
```
Comment (gpsanant) [inline cli/main.go:163]: "calls `verifyStaleBalances` to ~~correct the balance~~ start a checkpoint on a pod you don't own"

**Review by gpsanant — COMMENTED** [review]
This all looks good. Do we need a command for figuring out the slashed validator index though?

like, if another flag is set, it looks for a slashed validator and doesn't do anything if it doesn't find one, but proves if it does?

File: cli/main.go (hunk 1 above)
Comment (wadealexc) [inline cli/main.go:163]: Note that anything surrounded by backticks (eg \`verifyStaleBalances\` above) gets added as a "variable name"/"target" in the CLI help text. Would recommend not using these unless you're trying to change the help text

Run `./cli -h` to see what i'm talking about

File: cli/main.go [hunk 2]
```diff
@@ -128,6 +157,76 @@ func main() {
...
+						core.PanicOnError("failed to generate checkpoint proofs", err)
+
+						txns, err := core.SubmitCheckpointProof(ctx, sender, eigenpodAddress, chainId, proofs, eth, batchSize, noPrompt)
+						core.PanicOnError("failed to submit checkpoint proofs", err)
+
+						for i, txn := range txns {
+							fmt.Printf("sending txn[%d/%d]: %s (waiting)...", i, len(txns), txn.Hash())
+							bind.WaitMined(ctx, eth, txn)
+						}
+					}
```
Comment (wadealexc) [inline cli/main.go:202]: This could be quite expensive - should we add a flag that signals that you want to complete an existing checkpoint if it's there?

Comment (wadealexc) [reply cli/main.go:202]: like i'm picturing a worst-case scenario where we run the health checker on Renzo's pods and out of 8k validators one got slashed

and then we auto-complete a checkpoint for them

which, at 15 gwei, is ballpark ~6/7 ETH. Then, we start a checkpoint for them and complete a second one, for another 6/7 ETH :sweat:

Comment (jbrower95) [reply cli/main.go:202]: `submitCheckpointProof` has warning messages all over it (see how we pass `noPrompt` through). We should indicate ahead of time that _starting_ the checkpoint could lead to finishing an expensive one though..

but that's sort of the same problem we face on checkpoints rn in the CLI -- we don't do any cost/gas estimation yet

**Comment by jbrower95** [comment]
going to rebase / reopen this.

**Pull request closed without merging**